### PR TITLE
[ticket/14559] Remove attachment BBCode tags from the quoted message

### PR DESCRIPTION
**Pull request by rxu**
<a href="https://tracker.phpbb.com/browse/PHPBB3-14559">PHPBB3-14559</a>.

### CODE REVIEW

**Comment by rxu** [comment]
@JoshyPHP Not sure but I guess for 3.2 it should be handled in different way. I couldn't found the relevant textformatter util though (just the one which is removing the whole bbcode tag with its content).

**Comment by JoshyPHP** [comment]
@rxu Yes, since you remove the markup after the message has been decoded, it works the same way in 3.1 and 3.2.

File: phpBB/posting.php [hunk 1]
```diff
@@ -1589,6 +1589,9 @@
 
 if ($generate_quote)
 {
+	// Remove attachment bbcode tags from the quoted message to avoid mixing with the new post attachments if any
+	$message_parser->message = preg_replace('#\[attachment=([0-9]+)\](.*?)\[\/attachment\]#uis', '\\2', $message_parser->message);
```
Comment (Nicofuma) [inline phpBB/posting.php]: why do you keep the attachment description?

Comment (rxu) [inline phpBB/posting.php]: This is the file name unlinked, to keep the discussion context I'd say. And it's the same way it looks when you quote attachments without adding new ones to new post.

Comment (Nicofuma) [inline phpBB/posting.php]: ok

**Pull request merged**